code/modules/atmospherics/machinery/components/unary_devices/thermomachine.dm

```diff
@@ -200,6 +221,17 @@
 	use_power(power_usage)
 	update_parents()
 
+/obj/machinery/atmospherics/components/unary/thermomachine/attackby(obj/item/assembly/signaler/anomaly/pyro, mob/living/user, params)
+	if(pyro_anomaly_core || !allows_cooling)
```

Comment:
If regular thermomachines can't get the core (due to ``allows_cooling``, then what's the point of its own refreshparts and such?

Reply:
After looking into possible alternatives I can say the only case where it is absolutely needed is in examine proc so you can distinguish between the thermomachines with the core as a addon and the anomalous thermomachine.

Reply:
I replaced them with several `istype(src, /obj/machinery/atmospherics/components/unary/thermomachine/anomalous)` let me know if the var is preferred.